src/Symfony/Component/HttpFoundation/EventStreamResponse.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+/**
+ * Represents a streaming HTTP response for sending server events
+ * as part of the Server-Sent Events (SSE) streaming technique.
+ *
+ * @see ServerEvent
+ *
+ * @author Yonel Ceruto <open@yceruto.dev>
+ *
+ * Example usage:
+ *
+ *     return new EventStreamResponse(function () {
+ *         while (true) {
+ *             yield new ServerEvent(time(), type: 'ping');
+ *
+ *             if (connection_aborted()) {
+ *                 break;
+ *             }
+ *
+ *             sleep(1);
+ *         }
+ *     });
+ */
+class EventStreamResponse extends StreamedResponse
+{
+    /**
+     * @param int $retry The event reconnection time in milliseconds
+     */
+    public function __construct(?callable $callback = null, int $status = 200, array $headers = [], private int $retry = 0)
+    {
+        $headers += [
+            'Content-Type' => 'text/event-stream',
+            'Cache-Control' => 'no-cache',
+            'Connection' => 'keep-alive',
+        ];
+
+        parent::__construct($callback, $status, $headers);
+    }
+
+    public function setCallback(callable $callback): static
+    {
+        if ($this->callback) {
+            return parent::setCallback($callback);
+        }
+
+        $this->callback = function () use ($callback) {
+            if (is_iterable($events = $callback($this))) {
+                foreach ($events as $event) {
+                    $this->sendEvent($event);
+                }
+            }
+        };
+
+        return $this;
+    }
+
+    /**
+     * @param bool $flush Whether output buffers should be flushed
+     *
+     * @return $this
+     */
+    public function sendEvent(ServerEvent $event, bool $flush = true): static
```

Comment:
What is the use case of not flushing?

Reply:
I was thinking on sending multiple events together and reducing the number of separate writes (and thus TCP packets or system calls) but the benefit may be marginal, so I removed it.